jupyterhub/tests/conftest.py

```diff
@@ -322,26 +325,26 @@ def cleanup():
             with raises(TimeoutExpired):
                 service.proc.wait(1)
         if url:
-            io_loop.run_sync(partial(service.server.wait_up, http=True))
+            await service.server.wait_up(http=True)
     return service
 
 
 @fixture
-def mockservice(request, app):
+async def mockservice(request, app):
     """Mock a service with no external service url"""
-    yield _mockservice(request, app, url=False)
+    yield await _mockservice(request, app, url=False)
```

Comment:
I have not encountered a scenario where `yield` and `await` are used in this way. Can you please help me understand the usage?